mlx_lm/generate.py

```diff
@@ -783,30 +790,39 @@ def main():
         mx.random.seed(args.seed)
 
     # Load the prompt cache and metadata if a cache file is provided
+    cache_exists = False
     using_cache = args.prompt_cache_file is not None
     if using_cache:
-        prompt_cache, metadata = load_prompt_cache(
-            args.prompt_cache_file,
-            return_metadata=True,
+        try:
+            prompt_cache, metadata = load_prompt_cache(
```

Comment:
I would change this logic a little. If a prompt cache file is provided at least one of the following must be true: either you found an existing file and loaded from it, or you set `save_prompt_cache`. If neither is true, then it would be better to raise an error than continue silently.